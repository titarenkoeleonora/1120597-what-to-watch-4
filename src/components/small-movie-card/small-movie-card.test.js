import React from "react";
import renderer from "react-test-renderer";
import SmallMovieCard from "./small-movie-card";
import {Router} from "react-router-dom";
import history from "../../history";

const testMovie = {
  id: 0,
  title: `Movie title`,
  poster: ``,
  previewImage: ``,
  bgImage: ``,
  backgroundColor: ``,
  src: ``,
  previewVideoLink: ``,
  description: ``,
  rating: 8.2,
  votes: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`],
  runTime: 99,
  genre: `Genre`,
  date: 2020,
  isFavorite: true,
};

it(`Small movie card correctly render`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <SmallMovieCard
            movie={testMovie}
            isPlaying={false}
            onSmallCardMouseOver={() => {}}
            onSmallCardMouseOut={() => {}}
            onMovieCardClick={() => {}}
          />
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        })
        .toJSON();

  expect(tree).toMatchSnapshot();
});
