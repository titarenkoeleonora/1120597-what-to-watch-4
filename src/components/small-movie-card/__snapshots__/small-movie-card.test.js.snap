// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small movie card correctly render 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster=""
      src=""
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/0"
      onClick={[Function]}
    >
      Movie title
    </a>
  </h3>
</article>
`;
